### PR TITLE
Get app by local tenant id and app template id

### PR DESCRIPTION
**Pull request by yordanovagabriela**
**Description**

Changes proposed in this pull request:
- Introduce a new graphql request to get application by local tenant id and app template id

**Related issue(s)**

- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/director/config.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ graphql:
     applications: ["application:read"]
     application: ["application:read"]
     applicationBySystemNumber: ["application:read"]
+    applicationByLocalTenantIDAndAppTemplateID: ["application:read"]
```
Comment (kaloyanspiridonov) [inline chart/compass/charts/director/config.yaml:9]: add to director/hack/config-local.yaml as well

File: components/director/internal/domain/root_resolver.go [hunk 2]
```diff
@@ -495,6 +495,11 @@ func (r *queryResolver) ApplicationBySystemNumber(ctx context.Context, systemNum
 	return r.app.ApplicationBySystemNumber(ctx, systemNumber)
 }
 
+// ApplicationByLocalTenantIDAndAppTemplateID returns an application retrieved by local tenant id and app template id
+func (r *queryResolver) ApplicationByLocalTenantIDAndAppTemplateID(ctx context.Context, localTenantID, appTemplateID string) (*graphql.Application, error) {
```
Comment (kaloyanspiridonov) [inline components/director/internal/domain/root_resolver.go:499]: missing e2e tests

**Review by kaloyanspiridonov — CHANGES_REQUESTED** [review]

File: components/director/hack/config-local.yaml [hunk 3]
```diff
@@ -168,6 +168,7 @@ clientCredentialsRegistrationScopes:
     - "webhook:write"
   application:
     - "application:read"
+    - "applicationByLocalTenantIDAndAppTemplateID:read"
```
Comment (kaloyanspiridonov) [inline components/director/hack/config-local.yaml:171]: You are not introducing new scope. You should add "applicationByLocalTenantIDAndAppTemplateID: ["application:read"]" in the query section

**Review by kaloyanspiridonov — CHANGES_REQUESTED** [review]

**Review by kaloyanspiridonov — APPROVED** [review]

**Review by ekaterina-mitova — APPROVED** [review]
Looks good to me.

**Pull request merged**